lazyflow/utility/io_util/OMEZarrStore.py

```diff
@@ -0,0 +1,181 @@
+            #   {'name': 'x', 'type': 'space', 'unit': 'nanometer'}
+            # ]
+            axis_keys = [d["name"] for d in ome_axes]
+    return vigra.defaultAxistags("".join(axis_keys))
+
+
+class OMEZarrStore(MultiscaleStore):
+    """
+    Adapter class to handle communication with a source serving a dataset in OME-Zarr format.
+
+    :param url: URL to the OME-Zarr store.
+    :param last_scale_only_mode:
+        If True, only the last scale is loaded to determine the dtype. Used to shorten init time
+        when DatasetInfo instantiates a standalone OpInputDataReader to get lane shape and dtype.
+    """
+
+    NAME = "OME-Zarr"
+    URL_HINT = 'URL contains "zarr"'
+
+    spec_schema = {
+        "type": "object",
+        "properties": {
+            "multiscales": {
+                "type": "array",
+                "items": {
+                    "type": "object",
+                    "properties": {
+                        "axes": {"type": "array"},
+                        "datasets": {
+                            "type": "array",
+                            "items": {
+                                "type": "object",
+                                "properties": {"path": {"type": "string"}},
+                                "required": ["path"],
+                            },
+                        },
+                    },
+                    "required": ["datasets"],
+                },
+            },
+        },
+        "required": ["multiscales"],
+    }
+
+    def __init__(self, url: str = "", last_scale_only_mode: bool = False):
+        with Timer() as timer:
+            self.url = url
+            uncached_store = FSStore(self.url, mode="r", **OME_ZARR_V_0_4_KWARGS)
+            try:
+                self.ome_spec = json.loads(uncached_store[".zattrs"])
+            except KeyError:
+                raise ValueError("Expected a Zarr store, but could not find .zattrs file at the address.")
+            if _get_ome_spec_version(self.ome_spec) == "0.1":
+                uncached_store = FSStore(self.url, mode="r", **OME_ZARR_V_0_1_KWARGS)
+            self._store = LRUStoreCache(uncached_store, max_size=None)
+            logger.info(f"Initializing OME-Zarr store at {url} took {timer.seconds()*1000} ms.")
+        try:
+            jsonschema.validate(self.ome_spec, self.spec_schema)
+        except jsonschema.ValidationError:
+            raise ValueError(f"Metadata for this dataset did not match OME-Zarr spec.\nReceived:\n{self.ome_spec}.")
+        multiscale_spec = self.ome_spec["multiscales"][0]
+        axistags = _get_axistags_from_spec(multiscale_spec)
+        datasets = multiscale_spec["datasets"]
+        assert len(datasets) > 0, "The OME-Zarr store contains no datasets."
+        dtype = None
+        gui_scale_metadata = {}  # Becomes slot metadata -> must be serializable (no ZarrArray allowed)
+        self._scale_data = {}
+        for scale in reversed(datasets):
```

Comment:
Using `OrderedDict` from `collections` would slightly improve understanding of the code where normally I don't think users rely much on the insertion order to dicts, and (at least it looks like it for me, since you reverse the sequence before inserting) you might.

Second point here concerns the reversing of order. I checked with both precomputed and ngff, and those seem to have the ordering from highest resolution (largest image shape) to lowest (smallest image shape). You seem to reverse this order, without documenting why (other than that maybe the dropdown menu starts with the lowest res, which would be the cheapest to load?!). If there is no good reason to reverse then I think sticking to the order that the majority of the data formats exhibit would be preferable...

Reply:
Reversing is indeed for the dropdown box, will move this to the gui and keep everything as ordered in the dataset till then